otcextensions/sdk/auto_scaling/v1/_proxy.py

```diff
@@ -448,6 +508,69 @@ def batch_instance_action(
         instance = _instance.Instance(scaling_group_id=group.id)
         return instance.batch_action(self, instances, action, delete_instance)
 
+    def wait_for_instance(self, instance, status='INSERVICE', failures=None,
+                          interval=2, wait=180):
+        """Wait for an instance to be in a particular status.
+
+        :param instance:
+            The :class:`~otcextensions.sdk.auto_scaling.v1.instance.Instance`
+            or ID to wait on to reach the specified status.
+        :param status: Desired status.
+        :param failures:
+            Statuses that would be interpreted as failures.
+        :type failures: :py:class:`list`
+        :param int interval:
+            Number of seconds to wait before to consecutive checks.
+            Default to 2.
+        :param int wait:
+            Maximum number of seconds to wait before the change.
+            Default to 180
+        :return: The resource is returned on success.
+        :raises: :class:`~openstack.exceptions.ResourceTimeout` if transition
+                 to the desired status failed to occur in specified seconds.
+        :raises: :class:`~openstack.exceptions.ResourceFailure` if the resource
+                 has transited to one of the failure statuses.
+        """
+        instance = self._get_resource(_instance.Instance, instance)
```

Comment:
In principle also here should the same as above work

Reply:
Done